src/move/emojicoin_dot_fun/sources/emojicoin_dot_fun.move

```diff
@@ -411,7 +447,17 @@ module emojicoin_dot_fun::emojicoin_dot_fun {
         let (supply_minuend, reserves_ref) =
             assign_supply_minuend_reserves_ref(market_ref_mut, in_bonding_curve);
         let circulating_supply = supply_minuend - reserves_ref.base;
-        let user_emojicoin_balance = coin::balance<Emojicoin>(user_address);
+        let user_emojicoin_balance = if (!coin::is_account_registered<Emojicoin>(user_address)) {
+            coin::register<Emojicoin>(user);
+            0
+        } else {
+            coin::balance<Emojicoin>(user_address)
+        };
+        let balance_as_fraction = if (circulating_supply == 0) {
```

Comment:
Highlighting this to flag it @alnoki , this was giving me divide by 0 error. It's possible I just didn't initialize things correctly in my unit tests, but not sure.

Reply:
It's possible that denominator is 0 if no coins are in circulation yet

e.g. you register market then immediately chat

should probably just escape that case, I'll catch it in tests